solutions/tests/test_count_words.py

```diff
@@ -0,0 +1,67 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""A module containing tests for the count_words function in solutions/count_words.py
+
+Module contents:
+    TestCountWords: A class that contains tests for the count_words function.
+
+Created on Sunday, 29th December, 2024.
+@author: Gai Samuel
+"""
+
+import unittest
+
+from solutions.count_words import count_words
+
+
+class TestCountWords(unittest.TestCase):
+    """Tests for count_words function"""
+
+    def test_count_words(self):
+        """count_words returns the correct word count for each word in the text"""
+        self.assertEqual(
+            count_words("My name is Gai"), {"my": 1, "name": 1, "is": 1, "gai": 1}
+        )
+
+    def test_count_words_with_an_empty_string(self):
+        """count_words returns an empty dictionary for an empty string"""
+        self.assertEqual(count_words(""), {})
+
+    def test_count_words_with_punctuation(self):
+        """Throws an assertion error if the punctuation is declared in it's own string"""
```

Comment:
I didn't understand this.

Reply:
If the text was Tom! and you enter "tom": 1, "!": 1 it gives an assertion error.